imu_filter_madgwick/src/imu_filter_ros.cpp

```diff
@@ -21,149 +21,184 @@
 
   std::string world_frame;
-  if (!nh_private_.getParam ("world_frame", world_frame))
-    world_frame = "enu";
-
+  declare_parameter("world_frame", "enu");
+  get_parameter("world_frame", world_frame);
   if (world_frame == "ned") {
     world_frame_ = WorldFrame::NED;
-  } else if (world_frame == "nwu"){
+  } else if (world_frame == "nwu") {
     world_frame_ = WorldFrame::NWU;
-  } else if (world_frame == "enu"){
+  } else if (world_frame == "enu") {
     world_frame_ = WorldFrame::ENU;
   } else {
-    ROS_ERROR("The parameter world_frame was set to invalid value '%s'.", world_frame.c_str());
-    ROS_ERROR("Valid values are 'enu', 'ned' and 'nwu'. Setting to 'enu'.");
+    RCLCPP_ERROR(get_logger(), "The parameter world_frame was set to invalid value '%s'.",
+                 world_frame.c_str());
+    RCLCPP_ERROR(get_logger(), "Valid values are 'enu', 'ned' and 'nwu'. Setting to 'enu'.");
     world_frame_ = WorldFrame::ENU;
   }
   filter_.setWorldFrame(world_frame_);
 
   // check for illegal constant_dt values
   if (constant_dt_ < 0.0)
   {
-    ROS_FATAL("constant_dt parameter is %f, must be >= 0.0. Setting to 0.0", constant_dt_);
+    RCLCPP_ERROR(get_logger(), "constant_dt parameter is %f, must be >= 0.0. Setting to 0.0",
+                 constant_dt_);
     constant_dt_ = 0.0;
   }
 
   // if constant_dt_ is 0.0 (default), use IMU timestamp to determine dt
   // otherwise, it will be constant
-  if (constant_dt_ == 0.0)
-    ROS_INFO("Using dt computed from message headers");
-  else
-    ROS_INFO("Using constant dt of %f sec", constant_dt_);
+  if (constant_dt_ == 0.0) {
+    RCLCPP_INFO(get_logger(), "Using dt computed from message headers");
+  } else {
+    RCLCPP_INFO(get_logger(), "Using constant dt of %f sec", constant_dt_);
+  }
 
-  if (remove_gravity_vector_)
-    ROS_INFO("The gravity vector will be removed from the acceleration");
-  else
-    ROS_INFO("The gravity vector is kept in the IMU message.");
+  if (remove_gravity_vector_) {
+    RCLCPP_INFO(get_logger(), "The gravity vector will be removed from the acceleration");
+  } else {
+    RCLCPP_INFO(get_logger(), "The gravity vector is kept in the IMU message.");
+  }
+
+  // TODO: Add reconfigure parameter descriptions.
+  // gen.add("gain", double_t, 0, "Gain of the filter. Higher values lead to faster convergence but
+  // more noise. Lower values lead to slower convergence but smoother signal.", 0.1, 0.0, 1.0)
+  // gen.add("zeta", double_t, 0, "Gyro drift gain (approx. rad/s).", 0, -1.0, 1.0)
+  // gen.add("mag_bias_x", double_t, 0, "Magnetometer bias (hard iron correction), x component.", 0,
+  // -10.0, 10.0) gen.add("mag_bias_y", double_t, 0, "Magnetometer bias (hard iron correction), y
+  // component.", 0, -10.0, 10.0) gen.add("mag_bias_z", double_t, 0, "Magnetometer bias (hard iron
+  // correction), z component.", 0, -10.0, 10.0) gen.add("orientation_stddev", double_t, 0,"Standard
+  // deviation of the orientation estimate.", 0, 0, 1.0)
```

Comment:
Could you finish this TODO? Also, could you please delete the file `cfg/ImuFilterMadgwick.cfg`?

Reply:
Still working on this one.

Reply:
Great! Once you've finished this TODO and the other two above, I'll squash + merge. It would be great if you could quickly test the final version again before pushing, since I don't have the test setup at the moment.

Reply:
Please check latest update for the reconfigurable parameter descriptions. I could not find a better way in existing eloquent rclcpp to define the parameter descriptions. This method comes from the nav2 stack.

Reply:
We will definitely test since we are using this same code. I also have changes to add yaw offset and declination, but I did not want to post those until this was complete.